src/sa_basis.jl

```diff
@@ -0,0 +1,160 @@
+"""
+    isotypical_basis(χ::AbstractClassFunction)
+Compute a basis of the image of the projection corresponding to a class function `χ`.
+
+Return the coefficients of basis vectors in an invariant subspace corresponding to `χ`
+(so called _isotypical subspace_) in the action on `ℝⁿ` encoded by the conjugacy classes of `χ`.
+"""
+function isotypical_basis(χ::AbstractClassFunction)
+    u, weight = matrix_projection(χ)
+    image, pivots = if iszero(weight)
+        u_ = similar(u, 0, size(u, 2))
+        image_basis!(u_)
+    else
+        u .*= weight
+        image_basis!(u)
+    end
+    @debug "isotypical subspace corresponding to χ has dimension $(length(pivots))" χ
+
+    return image[pivots, :]
+end
+
+struct SemisimpleSummand{T}
+    basis::T
+    multiplicity::Int
+end
+
+basis(b::SemisimpleSummand) = b.basis
+Base.convert(::Type{M}, b::SemisimpleSummand) where {M<:AbstractMatrix} =
+    convert(M, basis(b))
+
+multiplicity(b::SemisimpleSummand) = b.multiplicity
+
+function degree(b::SemisimpleSummand)
+    d, r = divrem(size(basis(b), 1), multiplicity(b))
+    @assert iszero(r)
+    return d
+end
+
+function extended_characters(::Type{T}, G::Group, basis, action) where {T}
+    chars = characters_dixon(T, G)
+    ehom = ExtensionHomomorphism(action, basis)
+    return ehom(chars) # result: Cyclotomic{T} valued Characters
+end
+
+"""
+    symmetry_adapted_basis([T::Type,] G::AbstractPermutationGroup[, S=Rational{Int}])
+Compute a basis for the linear space `ℝⁿ` which is invariant under the symmetry of `G`.
+
+The permutation group is acting naturally on `1:degree(G)`. The coefficients of
+the invariant basis are returned in (orthogonal) blocks corresponding to irreducible
+characters of `G`.
+
+Arguments:
+* `S` controlls the types of `Cyclotomic`s used in the computation of
+character table. Exact type are preferred. For larger groups `G` `Rational{BigInt}`
```

Comment:
It might be better to use `Rational{BigInt}` by default so that if the user gets incorrect result with `Rational{Int}`, he's responsible

Reply:
By default julia uses checked operations with Rationals, so i don't think users can get silently invalid results only because we use `Rational{Int}`s. Therefore I'd keep it as it is